web/src/engine/websites/ComicFuz.ts

```diff
@@ -0,0 +1,179 @@
+    pages: ViewerPage[];
+
+}
+
+type ViewerPage = {
+    image: ApiImage
+}
+
+type ApiImage = {
+    imageUrl: string,
+    urlScheme: string,
+    iv: string,
+    encryptionKey: string,
+    imageWidth: number,
+    imageHeight: number,
+    isExtraPage: boolean
+}
+
+type MangasByDayOfWeekResponse = {
+    mangas: ApiManga[]
+}
+
+export default class extends DecoratableMangaScraper {
+
+    private readonly apiUrl = 'https://api.comic-fuz.com';
+    private readonly imgUrl = 'https://img.comic-fuz.com';
+
+    public constructor() {
+        super('comicfuz', 'COMIC FUZ', 'https://comic-fuz.com', Tags.Media.Manga, Tags.Language.Japanese, Tags.Source.Official);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return new RegExp(`^${this.URI.origin}/manga/\\d+$`).test(url);
+
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const id = new URL(url).pathname.split('/').pop();
+        const data = await this.fetchMangaDetail(id);
+        return new Manga(this, provider, id, data.manga.title);
+    }
+
+    private async fetchMangaDetail(titleId: string): Promise<MangaDetailResponse> {
+        const uri = new URL('v1/manga_detail', this.apiUrl);
+        const payload = {
+            deviceInfo: {
+                deviceType: 2,
+            },
+            mangaId: titleId
+        };
+        const request = await this.createPROTORequest(uri, 'ComicFuz.MangaDetailRequest', payload);
+        return FetchProto<MangaDetailResponse>(request, protoTypes, 'ComicFuz.MangaDetailResponse');
+    }
+
+    private async createPROTORequest(uri: URL, messageProtoType: string, payload: unknown): Promise<FetchRequest> {
+        const root = await protobuf.parse(protoTypes, { keepCase: true }).root;
+        const messageType = root.lookupType(messageProtoType);
+        const message = messageType.encode(payload);
+        return new FetchRequest(uri.href, {
+            body: message.finish(),
+            method: 'POST'
+        });
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const uri = new URL('v1/mangas_by_day_of_week', this.apiUrl);
+        const payload = {
+            deviceInfo: {
+                deviceType: 2
+            },
+            dayOfWeek: 0
+        };
+        const request = await this.createPROTORequest(uri, 'ComicFuz.MangasByDayOfWeekRequest', payload);
+        const data = await FetchProto<MangasByDayOfWeekResponse>(request, protoTypes, 'ComicFuz.MangasByDayOfWeekResponse');
+        return data.mangas.map(manga => new Manga(this, provider, manga.id.toString(), manga.title));
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const data = await this.fetchMangaDetail(manga.Identifier);
+        const chapters: Chapter[] = [];
+        data.chapters.forEach(chapterGroup => {
```

Comment:
May use `.reduce()` instead

Reply:
Not very familiar with reduce but i just found flat

```        return data.chapters.map(group => group.chapters.map(chapt => new Chapter(this, manga, chapt.id.toString(), chapt.title))).flat(); ``` is perfect